website/static/js/oauthAddonNodeConfig.js

```diff
@@ -0,0 +1,244 @@
+    },
+    importAuth: function(){
+        var self = this;
+
+        self.updateAccounts(function() {
+            if (self.accounts().length > 1) {
+                bootbox.prompt({
+                    title: 'Choose ' + self.addonName + ' Access Token to Import',
+                    inputType: 'select',
+                    inputOptions: ko.utils.arrayMap(
+                        self.accounts(),
+                        function(item) {
+                            return {
+                                text: item.name,
+                                value: item.id
+                            };
+                        }
+                    ),
+                    value: self.accounts()[0].id,
+                    callback: (self.connectExistingAccount.bind(self))
+                });
+            } else {
+                bootbox.confirm({
+                    title: 'Import ' + self.addonName + ' Access Token?',
+                    message: self.messages.confirmAuth(),
+                    callback: function(confirmed) {
+                        if (confirmed) {
+                            self.connectExistingAccount.call(self, (self.accounts()[0].id));
+                        }
+                    }
+                });
+            }
+        });
+    }
+});
+
+OauthAddonFolderPickerViewModel.prototype.connectExistingAccount = function(account_id) {
+     var self = this;
+
+    $osf.putJSON(
+        self.urls().importAuth, {
+            external_account_id: account_id
+        }
+    ).then(self.onImportSuccess.bind(self), self.onImportError.bind(self));
+};
+
+OauthAddonFolderPickerViewModel.prototype.updateAccounts = function(callback) {
+    var self = this;
+    var request = $.get(self.urls().accounts);
+    request.done(function(data) {
+        self.accounts(data.accounts.map(function(account) {
+            return {
+                name: account.display_name,
+                id: account.id
+            };
+        }));
+        callback();
+    });
+    request.fail(function(xhr, textStatus, error) {
+        self.changeMessage(self.messages.UPDATE_ACCOUNTS_ERROR(), 'text-warning');
+        Raven.captureMessage('Could not GET ' + self.addonName + ' accounts for user', {
+            url: self.url,
+            textStatus: textStatus,
+            error: error
+        });
+    });
+};
+
+// Public API
+function OauthAddonNodeConfig(addonName, selector, url, folderPicker, opts) {
+    var self = this;
+    self.url = url;
+    self.folderPicker = folderPicker;
+    if (typeof opts === 'undefined') {
+        opts = {};
+    }
+    self.viewModel = new OauthAddonFolderPickerViewModel(addonName, url, selector, folderPicker, opts);
+    self.viewModel.updateFromData();
+    $osf.applyBindings(self.viewModel, selector);
+}
+
+module.exports = {
+    OauthAddonNodeConfig: OauthAddonNodeConfig,
+    _OauthAddonNodeConfigViewModel: OauthAddonFolderPickerViewModel
+};
```

Comment:
Ensure that every file ends with a newline. (this can be done via an automatic setting in pycharm)